### PR TITLE
Cms Kit - Blogs Refactoring

### PR DESCRIPTION
**Pull request by enisn**

### CODE REVIEW

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs [hunk 1]
```diff
@@ -20,18 +20,20 @@ public class Blog : FullAuditedAggregateRoot<Guid>, IMultiTenant
...
-        public string Name { get; protected set; }
+        [NotNull] 
+        public virtual string Name { get; protected set; }
 
-        public string Slug { get; protected set; }
+        [NotNull] 
+        public virtual string Slug { get; protected set; }
 
-        public Guid? TenantId { get; protected set; }
+        public virtual Guid? TenantId { get; protected set; }
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs:29]: `TenantId` is optional so no need to make the setter `protected`.

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogPostManager.cs [hunk 2]
```diff
@@ -1,60 +1,65 @@
...
+        protected IBlogPostRepository BlogPostRepository { get; }
+        protected IBlogRepository BlogRepository { get; }
 
         public BlogPostManager(
             IBlogPostRepository blogPostRepository,
             IBlogRepository blogRepository)
         {
-            this.blogPostRepository = blogPostRepository;
-            this.blogRepository = blogRepository;
+            this.BlogPostRepository = blogPostRepository;
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogPostManager.cs]: no need to use `this`

File: modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Blogs/MongoBlogRepository.cs [hunk 3]
```diff
@@ -13,14 +14,14 @@ public MongoBlogRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbCont
         {
         }
 
-        public async Task<bool> ExistsAsync(Guid blogId)
+        public virtual async Task<bool> ExistsAsync(Guid blogId, CancellationToken cancellationToken = default)
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Blogs/MongoBlogRepository.cs:17]: Why don't we use the cancellation token inside the methods?

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ﻿using Microsoft.AspNetCore.Authorization;
 using System;
+using System.Diagnostics.CodeAnalysis;
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs]: Is this necessary?

**Review by ilkayilknur — APPROVED** [review]

**Review by ilkayilknur — CHANGES_REQUESTED** [review]
I left some small improvement comments. Rest LGTM

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs (hunk 1 above)
Comment (enisn) [reply modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs:29]: Should be TenantId able to change later?

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs (hunk 4 above)
Comment (enisn) [reply modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs]: Nope, removed.

**Review by ilkayilknur — APPROVED** [review]

**Pull request merged**